src/main/java/org/apache/commons/imaging/formats/jpeg/decoder/JpegDecoder.java

```diff
@@ -393,6 +418,74 @@ private void readMCU(final JpegInputStream is, final int[] preds, final Block[]
         }
     }
 
+    /**
+     * Returns an array of JpegInputStream where each field contains the JpegInputStream
+     * for one interval.
+     * @param scanPayload array to read intervals from
+     * @return JpegInputStreams for all intervals, at least one stream is always provided
+     */
+    static JpegInputStream[] splitByRstMarkers(final int[] scanPayload) {
+        final List<Integer> intervalStarts = getIntervalStartPositions(scanPayload);
+        // get number of intervals in payload to init an array of approbiate length
+        final int intervalCount = intervalStarts.size();
+        JpegInputStream[] streams = new JpegInputStream[intervalCount];
+        for (int i = 0; i < intervalCount; i++) {
+            int from = intervalStarts.get(i);
+            int to;
+            if (i < intervalCount - 1) {
+                // because each restart marker needs two bytes the end of
+                // this interval is two bytes before the next interval starts
+                to = intervalStarts.get(i + 1) - 2;
+            } else { // the last interval ends with the array
+                to = scanPayload.length;
+            }
+            int[] interval = Arrays.copyOfRange(scanPayload, from, to);
+            streams[i] = new JpegInputStream(interval);
+        }
+        return streams;
+    }
+
+    /**
+     * Returns the positions of where each interval in the provided array starts.
+     * @param scanPayload array to examine
+     * @return the start positions, the number of start positions is also the count of intervals
+     * while the number of restart markers found = number of start positions - 1
+     * because restart markers are between intervals
```

Comment:
What about moving the longer text to the main body of the javadocs comment? i.e.

```java
     * Returns the positions of where each interval in the provided array starts. The number
     * of start positions is also the count of intervals while the number of restart markers
     * found is equal to the number of start positions minus one (because restart markers
     * are between intervals).
     * @param scanPayload array to examine
     * @return the start positions
```